### PR TITLE
Improve source installation of python bindings.

### PR DESCRIPTION
**Pull request by kevinzakka**
Improve Python bindings source installation.

Add support for conda environments to `make_sdist.sh` and upgrade `setuptools` and `wheel` along with pip.

Add step by step instructions in the README with examples for macOS.

PS: I should probably add linux installation instructions to the README when I get a chance.

### CODE REVIEW

File: python/README.md [hunk 1]
```diff
@@ -27,15 +29,48 @@ pip install mujoco
...
-generate a [source distribution](https://packaging.python.org/en/latest/glossary/#term-Source-Distribution-or-sdist)
-tarball, then run `pip wheel name_of_sdist.tar.gz`. The `make_sdist.sh` script
-generates additional C++ header files that are needed to build the bindings,
-and also pulls in required files from elsewhere in the repository outside the
-`python` directory into the sdist.
+### Source
+
+**Note.** CMake and a C++17 compiler are needed to build the bindings from source.
+
+1. Download the [latest release](https://github.com/deepmind/mujoco/releases) from GitHub. On macOS, the download corresponds to a DMG file from which you can drag the `MuJoCo` app into your Applications folder.
```
[the remark this answers is not in the document]
Comment (kevinzakka) [reply python/README.md]: Is this clear enough?

File: python/README.md [hunk 2]
```diff
@@ -27,15 +29,49 @@ pip install mujoco
...
+
+```bash
+git clone https://github.com/deepmind/mujoco.git
+cd mujoco/python
+```
+
+3. Create a virtual environment:
+
+```bash
+python3 -m venv /tmp/mujoco && /tmp/mujoco/bin/activate
```
Comment (saran-t) [inline python/README.md:54]: Should be `&& source /tmp/mujoco/bin/activate`. My bad.

May be clearer to just put the second command it in a separate line.

File: python/README.md [hunk 3]
```diff
@@ -27,15 +29,49 @@ pip install mujoco
...
+
+5. Install the generated tarball. You'll need to specify the path to the MuJoCo library you downloaded earlier. For example, on macOS, this will be `/Applications/MuJoCo.app/Contents/Frameworks/MuJoCo.framework` if you dragged it to your Applications folder.
+
+```bash
+cd dist
+MUJOCO_PATH=/PATH/TO/MUJOCO pip install mujoco-2.1.X.tar.gz
+```
 
-CMake and a C++17 compiler are needed to build the bindings from source.
+The python bindings should now be installed! To check that they've been successfully installed, cd outside of the mujoco directory and run `python -c "import mujoco"`.
```
Comment (saran-t) [inline python/README.md:74]: nit: Python (capitalised)

Comment (saran-t) [inline python/README.md:74]: nit: `cd` and `mujoco` (add backticks)

**Review by saran-t — CHANGES_REQUESTED** [review]

**Review by saran-t — APPROVED** [review]

File: python/README.md [hunk 4]
```diff
@@ -27,15 +29,50 @@ pip install mujoco
...
+### Source
+
+**Note.** Building from source is only necessary if you are modifying the Python bindings (or are trying to run on exceptionally old Linux systems). If that's not the case, then we recommend installing the prebuilt binaries from PyPI.
+
+Before proceeding, make sure you have CMake and a C++17 compiler installed.
+
+1. Download the latest release of the [binary archives](https://github.com/deepmind/mujoco/releases) from GitHub. On macOS, the download corresponds to a DMG file from which you can drag the `MuJoCo` app into your Applications folder.
+
+<p align="center">
+  <img src="../assets/macos_install.png" width=600>
```
Comment (saran-t) [inline python/README.md:41]: Actually, can we delete this screenshot for now?

**Review by saran-t — CHANGES_REQUESTED** [review]

File: python/README.md (hunk 4 above)
Comment (kevinzakka) [reply python/README.md]: Done!

**Review by saran-t — APPROVED** [review]

**Review by saran-t — APPROVED** [review]

**Pull request merged**